Controller/CRUDController.php

```diff
@@ -1358,9 +1323,12 @@ protected function preList(Request $request)
      * To keep backwards compatibility with older Sonata Admin code.
      *
      * @internal
+     * @deprecated Since version 2.4, to be removed in 3.0
      */
     private function resolveRequest(Request $request = null)
     {
+        trigger_error(__CLASS__.'::'.__METHOD__.' is deprecated since 2.4 and will be removed in 3.0. Use $request passed directly on your controller action.');
```

Comment:
This method can be directly removed for two reasons:

* It's a private method
* This PR will be for the next major release